lib/oli/groups.ex

```diff
@@ -122,6 +122,35 @@ defmodule Oli.Groups do
   # ------------------------------------------------------------
   # Communities accounts
 
+  @doc """
+  Finds or creates a community account for a user. More efficient than using
+  many-to-many assoc.
+
+  ## Examples
+
+      iex> find_or_create_community_user_account(user_id, community_id)
+      {:ok, %CommunityAccount{}}
+
+      iex> find_or_create_community_user_account(bad_user_id, community_id)
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def find_or_create_community_user_account(user_id, community_id) do
+    case Repo.one(
```

Comment:
This "find" and "create" as two separate steps needs to be wrapped within a transaction.